pkg/resource/resource.go

```diff
@@ -190,13 +191,24 @@ func IsAPIErrorWrapped(err error) bool {
 	return IsAPIError(errors.Cause(err))
 }
 
+// IsNonConflictAPIErrorWrapped returns true if err is a non-conflict K8s API
+// error, or recursively wraps a K8s API error
+func IsNonConflictAPIErrorWrapped(err error) bool {
+	cause := errors.Cause(err)
+	return IsAPIError(cause) && !kerrors.IsConflict(cause)
+}
+
 // IsConditionTrue returns if condition status is true
 func IsConditionTrue(c xpv1.Condition) bool {
 	return c.Status == corev1.ConditionTrue
 }
 
-// An Applicator applies changes to an object.
+// An Applicator applies changes to an object. The passed object is expected to
+// be complete, i.e. not partial.
 type Applicator interface {
+	// Apply updates the given object to exactly the given state. It conflicts
+	// if the resource version stored on the apiserver does not match anymore
+	// the one in the given object.
```

Comment:
@turkenh this comment and the one above are crucial. Any other behaviour must go into an `ApplyOption`.